### ISSUE TITLE
htmlunit doesn't register event listener on 'readystatechange'

### ISSUE THREAD

**Issue by QIvan** [HtmlUnit/htmlunit#561]
Hi! First of all, thank you very much for this beautiful tool. It's extremely useful, and we use it all over the place in our projects. 

Last week we decided to upgrade it from 2.14 to the latest version (yeah-yeah, we didn't rush with it, I know :smile: )
but we noticed that now Htmlunit behaves a bit differently than any browser: 

We compose a small HTML: 
```html
<html>
    <body>
        <script>
            alert('Initial State:' + document.readyState);
            document.addEventListener('readystatechange', function() {
               alert('Changed:' + document.readyState);
            });
        </script>
    </body>
</html>
```

I kick it off  with python HTTP server like 
```bash 
python3 -m http.server
```

If I open http://localhost:8000 in my browser (I tested with chrome and firefox) 
I will see the next notifications: 
* Initial State:loading
* Changed:interactive
* Changed:complete

Now I try to "open" this page with htmlunit: 
```java 
public static void main(String[] args) throws Exception {
    final WebClient webClient = new WebClient(BrowserVersion.CHROME);
    WebClientOptions webClientOptions = webClient.getOptions();
    webClientOptions.setJavaScriptEnabled(true);
    webClient.setAlertHandler((AlertHandler) (page, message) -> System.out.println(message));

    Page page = webClient.getPage("http://localhost:8000");
    System.out.println("HTMLUnit Doc ready state: "
            + ((HtmlPage) page).getDocumentElement().getReadyState());

}
``` 

in the output I see just the following: 
```
Initial State:loading
HTMLUnit Doc ready state: complete
```

So, correct me if I'm wrong, but it seems like htmlunit didn't create a listener, or it did handle `readystatechange`  some other way as a browser does.

**Comment by rbri** [comment]
Hi @QIvan,

> First of all, thank you very much for this beautiful tool. It's extremely useful, and we use it all over the place in our projects.

Thanks a lot (at least if you use this for your business - think about sponsoring https://github.com/sponsors/rbri )

> Last week we decided to upgrade it from 2.14 to the latest version (yeah-yeah, we didn't rush with it, I know 😄 )

Right in time, version 3 is on the finish lane (https://twitter.com/HtmlUnit)

> Htmlunit behaves a bit differently than any browser

Did a quick check and i can reproduce your finding - thanks for the explanation / well done!
Will have a look...

Was this ever working in the old version?

**Comment by izemlyanskiy** [comment]
Thank you for a quick response.
> Right in time, version 3 is on the finish lane

Sweet! Will look forward some awesome new features =)

>Was this ever working in the old version?

Yes, with version 2.14 it worked just fine (I didn't test on my small example, but in our project this is the place where tests become red after upgrade)

> think about sponsoring

Good idea, will cosider that. 
Thanks!

**Comment by rbri** [comment]
> Yes, with version 2.14 it worked just fine (I didn't test on my small example, but in our project this is the place where tests become red after upgrade)

Strange - i have no idea why this got lost on the way.

**Comment by rbri** [comment]
> Sweet! Will look forward some awesome new features =)

Not so many but you have to change all package names... (org.htmlunit....)

**Comment by rbri** [comment]
I have a snapshot build ready with your fix.... (and all the changed package names)

There is a new snapshot repo at 

        <repository>
            <snapshots>
                <enabled>true</enabled>
                <updatePolicy>always</updatePolicy>
            </snapshots>
            <releases>
                <enabled>false</enabled>
            </releases>
            <id>S01 OSS Sonatype snapshots</id>
            <url>https://s01.oss.sonatype.org/content/repositories/snapshots/</url>
        </repository>
    </repository>

You have to change the dependency to

    <dependencies>
      <dependency>
          <groupId>org.htmlunit</groupId>
          <artifactId>htmlunit</artifactId>
          <version>3.0.0-SNAPSHOT</version>
      </dependency>
      
    </dependencies>

and finally adjust all your imports from com.gargoylesoftware.htmlunit... to org.htmlunit...

Hope this fixes your problem

**Comment by rbri** [comment]
Released with Version 3.0.0

**Comment by QIvan** [comment]
Thank you, @rbri sorry I missed your last comments (I had a few days off and tried to be afk as much as possible :smiley:)
I'll double-check with your last release later this week (atm we put a workaround in the tests)
